contrib/auto-render/auto-render.js

```diff
@@ -25,10 +25,11 @@ const renderMathInText = function(text, delimiters) {
         } else {
             const span = document.createElement("span");
             const math = data[i].data;
+            // Override any display mode defined in the settings with that
+            // defined by the text itself
+            renderOptions.displayMode = data[i].display;
```

Comment:
Also, perhaps there should be an `if (data[i].display !== null)` enclosing this code?  That way, an unspecified `display` value lets you use the caller's value.  Ah, but then the options copying would have to be done here, so maybe not worth it...